Microsoft.Toolkit/Helpers/ObjectStorage/IFileStorageHelper.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Toolkit.Helpers
+{
+    /// <summary>
+    /// Service interface used to store data in files and folders.
+    /// </summary>
+    public interface IFileStorageHelper
+    {
+        /// <summary>
+        /// Determines whether a file already exists.
+        /// </summary>
+        /// <param name="filePath">Key of the file (that contains object).</param>
+        /// <returns>True if a value exists.</returns>
+        Task<bool> FileExistsAsync(string filePath);
+
+        /// <summary>
+        /// Retrieves an object from a file.
+        /// </summary>
+        /// <typeparam name="T">Type of object retrieved.</typeparam>
+        /// <param name="filePath">Path to the file that contains the object.</param>
+        /// <param name="default">Default value of the object.</param>
+        /// <returns>Waiting task until completion with the object in the file.</returns>
+        Task<T> ReadFileAsync<T>(string filePath, T? @default = default);
+
+        /// <summary>
+        /// Retrieves all file listings for a folder.
+        /// </summary>
+        /// <param name="folderPath">The path to the target folder.</param>
+        /// <returns>A list of file names in the target folder.</returns>
+        Task<IList<string>> ReadFolderAsync(string folderPath);
+
+        /// <summary>
+        /// Saves an object inside a file.
+        /// </summary>
+        /// <typeparam name="T">Type of object saved.</typeparam>
+        /// <param name="filePath">Path to the file that will contain the object.</param>
```

Comment:
Will this path be created if it doesn't exist? We should specify those types of things in our 'contract' of the interface for folks wanting to implement this pattern.

Reply:
I would assume so, but that behavior should be up to the implementer. They could always throw an exception if they don't support that method of generating folders dynamically, and force the dev to call CreateFolderAsync explicitly before SaveFileAsync.

Reply:
Sure, it's just easy for an implementer to use `<inheritdoc/>` and then not specify what things do, so if we can at least suggest what should happen, then we give hints to the implementers to think about these types of scenarios and how they're handled.